symfit/core/fit.py

```diff
@@ -2029,7 +1919,108 @@ def __call__(self, *args, **kwargs):
+        """
+        if isinstance(func, sympy.Derivative):
+            new_func = func.expr.xreplace(func2vars)
+            new_variables = tuple(var.xreplace(func2vars)
+                                  for var in func.variables)
+            return partial_diff(new_func, *new_variables)
+        else:
+            return func.xreplace(func2vars)
+
+    # Inverse dict so we can turn functions back into vars in the end
+    functions_as_vars = dict((v, k) for k, v in model.vars_as_functions.items())
+    # Create the jacobian components. The `vars` here in the model_dict are
+    # always of the type D(y, a), but the righthand-side might still contain
+    # functions instead of vars depending on the value of `as_functions`.
+    jac = {}
+    for func, expr in model.function_dict.items():
+        for param in model.params:
+            target = D(func, param)
+            dfdp = expr.diff(param)
+            if as_functions:
+                jac[partial_subs(target, functions_as_vars)] = dfdp
+            else:
+                # Turn Function objects back into Variables.
+                dfdp = dfdp.subs(functions_as_vars, evaluate=False)
+                jac[partial_subs(target, functions_as_vars)] = dfdp
+    # Next lines are needed for the Hessian, where the components of model still
+    # contain functions instead of vars.
+    if as_functions:
+        jac.update(model)
+    else:
+        jac.update({y: expr.subs(functions_as_vars, evaluate=False)
+                    for y, expr in model.items()})
+    return CallableModel(jac)
+
+def hessian_from_model(model):
+    """
+    Build a :class:`~symfit.core.fit.CallableModel` representing the Hessian of
+    ``model``.
+
+    This function make sure the chain rule is correctly applied for
+    interdependent variables.
+
+    :param model: Any symbolical model-type.
+    :return: :class:`~symfit.core.fit.CallableModel` representing the Hessian
+        of ``model``.
+    """
+    jac_model = jacobian_from_model(model, as_functions=True)
+    return jacobian_from_model(jac_model)
+
+def leastsquares_from_model(model):
+    """
+    Creates an analytical Least-Squares model for ``model``, which can be used
+    to solve analytical least-squares problems.
+
+    .. note: Currently only for use by :class:`LinearLeastSquares`, as no
+        analytical sum is included.
+
+    :param model: Any symbolical model-type.
+    :return: :class:`~symfit.core.fit.GradientModel` representing the :math:`\chi^2`
+        of ``model``.
+    """
+    chi2_expr = sum(((model[y] - y)**2 / model.sigmas[y]**2)
+                    for y in model.dependent_vars)
+    chi2 = Variable()
+    chi2_dict = {chi2: chi2_expr}
+    # Update it with any leftover interdependent variables we might need.
+    chi2_dict.update({var: expr for var, expr in model.items()
+                     if var not in model.dependent_vars})
+    return GradientModel(chi2_dict)
```

Comment:
Why no HessianModel? We know it's going to be 0, but we might as well tell whoever is going to use this in the future.

Reply:
With https://github.com/tBuLi/symfit/pull/201/commits/e173162581cf9716f01a3105c545914e5caacca7 this is now also dead weight, correct?